### ISSUE TITLE
Incorrect path for cache directories on windows

### ISSUE THREAD

**Issue by johanneshardt** [scalameta/metals#5590]
### Describe the bug

Seems that this issue (#2451) has resurfaced: 
![image](https://github.com/scalameta/metals/assets/46093733/99e507e0-bb16-4d42-9e5f-ddd8b5c672dd)

Steps to reproduce:
 - Minimal install: I've removed, sbt, coursier and scala-cli (only Metals installed).
 - Removed everything relevant in %APPDATA%, %USERPROFILE% (metals, ammonite, coursier, bloop folders etc.)
 - Created a build.sbt with only scalaVersion := "3.3.0" and main.scala
 - Set JAVA_HOME = C:\Program Files\OpenJDK\jdk-19.0.1
 - Opened VS Code in that directory

The "null" folder is created as soon as metals starts, the "powershell" folder is created when clicking "import build' on the popup from metals. 

Output when running `ProjectDirectories.from()` as described in #2451:
<img width="460" alt="image" src="https://github.com/scalameta/metals/assets/46093733/e318ebf5-b2ba-4d21-bbc0-b69509133711">

Setting env vars (COURSIER_BIN_DIR, COURSIER_CACHE, COURSIER_JVM_CACHE) as was recommended at some time in the past makes no difference for me. Building with `scala-cli` instead of `sbt`, by running `scala-cli setup-ide .` before opening VS Code makes no difference either.

This seems to be an issue with `directories-jvm` again (https://github.com/dirs-dev/directories-jvm/issues/51, https://github.com/dirs-dev/directories-jvm/issues/49), so I'm not sure what can be done on metals' side. Mostly posting this in case someone else runs into this and only finds #2451.

### Operating system

Windows

### Editor/Extension

VS Code

### Version of Metals

v1.24.0

**Comment by tgodzik** [comment]
I wonder if in such case we should do a workaround and replace `null` with `%APPDATA%` value and resolving `.cache` on it. That's the expected place, right? Alternatively, we could try `%APPDATA%/metals-config` to not interfere with possible existing cache.

Not the pretties workaround, but should work for us. What do you think?

**Comment by tgodzik** [comment]
Alternatively, we could just return `None` if the directory starts with `null/`

**Comment by tgodzik** [comment]
Let's try https://github.com/scalameta/metals/pull/5595 -> at least we will not create the null directory

**Comment by johanneshardt** [comment]
Seems reasonable 👍

**Comment by tgodzik** [comment]
I changed the fix a bit, so do let us know if that is not fixed in the end.

**Comment by johanneshardt** [comment]
Yes, that seems to work on my PC. Reverting metals to 1.0.1 reintroduced the bug after a restart as well. Thanks for the fix!

**Comment by NoahELE** [comment]
I found that the issue persists to exist on Windows. Importing Metals build keeps creating `null` and `PowerShell 7.4.5` folders under the project dir.

**Comment by tgodzik** [comment]
That's unexpected, but best to create new issue.

**Comment by NoahELE** [comment]
https://github.com/scalameta/metals/issues/6885

New issue created.
